olive/passes/pytorch/tensor_parallel.py

```diff
@@ -0,0 +1,101 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+# Automatically distribute PyTorch model using Tensor Parallelism
+# --------------------------------------------------------------------------
+
+import logging
+from abc import abstractmethod
+from pathlib import Path
+from typing import Any, Callable, ClassVar, Dict
+
+import torch
+from pydantic import validator
+
+from olive.hardware.accelerator import AcceleratorSpec
+from olive.model import DistributedPyTorchModel, PyTorchModel
+from olive.passes import Pass
+from olive.passes.olive_pass import PassConfigParam
+
+logger = logging.getLogger(__name__)
+
+
+class PyTorchTensorParallel(Pass):
+    RANKED_MODEL_NAME_FORMAT: ClassVar[str] = "model_{:02d}"
+
+    @staticmethod
+    def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigParam]:
+        # Note : The default world_size should be the no of gpus (AceleratorSpec.Device == GPU)
+        # in the target OliveSystem
+        return {
+            "world_size": PassConfigParam(
+                type_=int,
+                default=2,
+                required=True,
+                description=("Number of GPU nodes to distribute the model for. Must be greater than 1."),
+            ),
+        }
+
+    @staticmethod
+    def _validate_world_size(v):
+        if int(v) < 2:
+            raise ValueError("world_size should be >= 2")
+
+        return v
+
+    @staticmethod
+    def _validators() -> Dict[str, Callable]:
+        return {
+            "validate_distributor_config": validator("world_size", allow_reuse=True)(
+                PyTorchTensorParallel._validate_world_size
+            )
+        }
+
+    def _run_for_config(
+        self, model: PyTorchModel, data_root: str, config: Dict[str, Any], output_model_path: str
+    ) -> DistributedPyTorchModel:
+        self.world_size = int(config["world_size"])
```

Comment:
Since we want everything for a run to be configurable using the `config` dict and don't dynamically update attributes of a pass object if possible (want the pass to be independent of the model/run), would it be possible to pass `config["world_size"]` as arguments to whatever method that needs it? Like `split_weights` and `load_rank_weights`?

Reply:
Done!